From 44a6e7cd343ed4da9763ed8644baed1a6e406dc6 Mon Sep 17 00:00:00 2001
From: Joel Williams <joel.williams@codethink.co.uk>
Date: Mon, 20 Jun 2022 09:40:52 +0100
Subject: [PATCH] Add LED control for sifive unmatched

Added support for D12 PWM LED and D2 RGB LEDs for
sifive unmatche boards and removed A00 from model
by making changes to /arch/riscv/boot/dts/hifive-unmatched-a00.dts.
---
 .../boot/dts/sifive/hifive-unmatched-a00.dts  | 40 ++++++++++++++++++-
 1 file changed, 39 insertions(+), 1 deletion(-)

diff --git a/arch/riscv/boot/dts/sifive/hifive-unmatched-a00.dts b/arch/riscv/boot/dts/sifive/hifive-unmatched-a00.dts
index c4ed9efdff03..181e0b53ca55 100644
--- a/arch/riscv/boot/dts/sifive/hifive-unmatched-a00.dts
+++ b/arch/riscv/boot/dts/sifive/hifive-unmatched-a00.dts
@@ -3,13 +3,14 @@
 
 #include "fu740-c000.dtsi"
 #include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/leds/common.h>
 #include <dt-bindings/interrupt-controller/irq.h>
 
 /* Clock frequency (in Hz) of the PCB crystal for rtcclk */
 #define RTCCLK_FREQ		1000000
 
 / {
-	model = "SiFive HiFive Unmatched A00";
+	model = "SiFive HiFive Unmatched";
 	compatible = "sifive,hifive-unmatched-a00", "sifive,fu740-c000",
 		     "sifive,fu740";
 
@@ -25,6 +26,43 @@ memory@80000000 {
 		device_type = "memory";
 		reg = <0x0 0x80000000 0x4 0x00000000>;
 	};
+	pwmleds {
+		compatible = "pwm-leds";
+		green-d12 {
+			label = "green:d12";
+			color = <LED_COLOR_ID_GREEN>;
+			pwms = <&pwm0 0 7812500 PWM_POLARITY_INVERTED>;
+			active-low = <1>;
+			max-brightness = <255>;
+			linux,default-trigger = "none";
+		};
+		green-d2 {
+			label = "green:d2";
+			color = <LED_COLOR_ID_GREEN>;
+			pwms = <&pwm0 1 7812500 PWM_POLARITY_INVERTED>;
+			active-low = <1>;
+			max-brightness = <255>;
+			linux,default-trigger = "none";
+		};
+
+		red-d2 {
+			label = "red:d2";
+			color = <LED_COLOR_ID_RED>;
+			pwms = <&pwm0 2 7812500 PWM_POLARITY_INVERTED>;
+			active-low = <1>;
+			max-brightness = <255>;
+			linux,default-trigger = "none";
+		};
+
+		blue-d2 {
+			label = "blue:d2";
+			color = <LED_COLOR_ID_BLUE>;
+			pwms = <&pwm0 3 7812500 PWM_POLARITY_INVERTED>;
+			active-low = <1>;
+			max-brightness = <255>;
+			linux,default-trigger = "none";
+		};
+	};
 
 	hfclk: hfclk {
 		#clock-cells = <0>;
-- 
2.30.2

